### ISSUE TITLE
Release artifacts (TRM model, datasets) on Hugging Face

### ISSUE THREAD

**Issue by NielsRogge** [SamsungSAILMontreal/TinyRecursiveModels#2]
Hi @AlexiaJM 🤗

Niels here from the open-source team at Hugging Face. I discovered your work through Hugging Face's daily papers as yours got featured: https://huggingface.co/papers/2510.04871.
The paper page lets people discuss about your paper and lets them find artifacts about it (your models, datasets or demo for instance), you can also claim
the paper as yours which will show up on your public profile at HF, add Github and project page URLs.

Your abstract mentions that "Our code and weights are available at this https URL", and it's great to see the code for the Tiny Recursive Model (TRM) released on GitHub. It would be fantastic to also make the pre-trained TRM model checkpoints and the prepared datasets (for ARC-AGI, Sudoku-Extreme, and Maze-Hard) available on the 🤗 hub, to improve their discoverability/visibility.
We can add tags so that people find them when filtering https://huggingface.co/models and https://huggingface.co/datasets.

## Uploading models

See here for a guide: https://huggingface.co/docs/hub/models-uploading.

In this case, we could leverage the [PyTorchModelHubMixin](https://huggingface.co/docs/huggingface_hub/package_reference/mixins#huggingface_hub.PyTorchModelHubMixin) class which adds `from_pretrained` and `push_to_hub` to any custom `nn.Module`. Alternatively, one can leverages the [hf_hub_download](https://huggingface.co/docs/huggingface_hub/en/guides/download#download-a-single-file) one-liner to download a checkpoint from the hub.

We encourage researchers to push each model checkpoint to a separate model repository, so that things like download stats also work. We can then also link the checkpoints to the paper page.

## Uploading dataset

Would be awesome to make the datasets available on 🤗 , so that people can do:

```python
from datasets import load_dataset

dataset = load_dataset("your-hf-org-or-username/your-dataset")
```
See here for a guide: https://huggingface.co/docs/datasets/loading.

Besides that, there's the [dataset viewer](https://huggingface.co/docs/hub/en/datasets-viewer) which allows people to quickly explore the first few rows of the data in the browser.

Let me know if you're interested/need any help regarding this!

Cheers,

Niels
ML Engineer @ HF 🤗

**Comment by vedantdere** [comment]
Anyone able to reproduce the metrics ?
I am trying but the metrics coming are very low

**Comment by anthonymichelandroid-netizen** [comment]
> The reproduction by the ARC-AGI team is out! https://x.com/arcprize/status/1978872653109956883
> 
> The models are here: https://huggingface.co/arcprize/trm_arc_prize_verification

Thank you for the snapshots. 
Could we share a snippet code to initialize the model from the snapshot and make a simple test for one or a few simple test cases in json format ?

**Comment by bdsaglam** [comment]
> Anyone able to reproduce the metrics ? I am trying but the metrics coming are very low

Yes, I could reproduce it by both training it from scratch myself and also from checkpoints shared on HF. I initially evaluated on small subset of ARC-AGI 1 test set and was getting low scores (3-4%). I found out that this was because evaluator was using [the size of whole test set in denominator when calculating metrics.](https://github.com/SamsungSAILMontreal/TinyRecursiveModels/blob/main/evaluators/arc.py#L175) 
Not sure if this helps you.
